core/src/main/resources/error/README.md

```diff
@@ -0,0 +1,79 @@
+# Guidelines
+
+To throw a standardized exception, developers should use an error class and message parameters
+rather than an arbitrary error message.
+
+## Usage
+
+To throw an exception, do the following.
+
+1. Check if an appropriate error class already exists in `error-class.json`.
```

Comment:
I have a concern that this will be another burden. E.g. naming a error class, finding error class.

I also worry that the number of error classes will grow quickly and hard to maintain.

Reply:
Today, the error messages thrown from Spark are distributed across the entire code base with no single source of truth - as a result, it is hard to audit error message for quality and redundancy. With error classes, I'm hoping that we can improve the auditing process. However, I do recognize that the number of error classes will likely grow quickly, and that maintaining a high level of quality will require vigilant pruning. What do you think will help reduce the burden here?

Reply:
I think json file is more general for auditing, doc generation, internationalization (each error message has translations for different languages).